drain3/template_miner.py

```diff
@@ -173,21 +173,30 @@ def match(self, log_message: str, full_search_strategy="never") -> LogCluster:
     def get_parameter_list(self, log_template: str, content: str):
         escaped_prefix = re.escape(self.config.mask_prefix)
         escaped_suffix = re.escape(self.config.mask_suffix)
-        template_regex = re.sub(escaped_prefix + r".+?" + escaped_suffix, self.drain.param_str, log_template)
-        if self.drain.param_str not in template_regex:
-            return []
-        template_regex = re.escape(template_regex)
-        template_regex = re.sub(r'\\ +', r'\\s+', template_regex)
-        template_regex = "^" + template_regex.replace(escaped_prefix + r"\*" + escaped_suffix, "(.*?)") + "$"
+        content_masked =  re.sub(escaped_prefix + r".+?" + escaped_suffix, lambda match: self.config.mask_prefix + "*" + self.config.mask_suffix, content)
```

Comment:
It is assumed that it is safe to place masks inside the log message without changing the behavior of masking-patterns.

As masks are later used to store parameters, the first step is to replace the content of any existing masks in the message.